src/app.rs

```diff
@@ -672,17 +673,21 @@ fn retrieve_dynein_file_path(dft: DyneinFileType) -> Result<String, DyneinConfig
 }
 
 fn retrieve_or_create_dynein_dir() -> Result<String, DyneinConfigError> {
-    match dirs::home_dir() {
-        None => Err(DyneinConfigError::HomeDir),
-        Some(home) => {
-            let dir = format!("{}/{}", home.to_str().unwrap(), CONFIG_DIR);
-            if !path::Path::new(&dir).exists() {
-                debug!("Creating dynein config directory: {}", dir);
-                fs::create_dir(&dir)?;
-            };
-            Ok(dir)
-        }
-    }
+    let path = env::var(CONFIG_PATH_ENV_VAR_NAME).unwrap_or(
+        dirs::home_dir()
+            .ok_or(DyneinConfigError::HomeDir)?
+            .to_str()
+            .ok_or(DyneinConfigError::HomeDir)?
+            .to_string(),
+    );
+
+    let dir = format!("{}/{}", path, CONFIG_DIR);
```

Comment:
Could you use [`path::join`](https://doc.rust-lang.org/std/path/struct.Path.html#method.join)? If you specify like `export DYNEIN_CONFIG_DIR=~/.cnf/`, `dir` become like `~/.cnf//.dynein/cache.yml`. I'm not aware the specific problem regarding this a path, but I think we should avoid this type of path to prevent future problem.